pkg/engine/funcs.go

```diff
@@ -175,3 +177,36 @@ func fromJSONArray(str string) []interface{} {
 	}
 	return a
 }
+
+// jq is like sed for JSON/YAML data - you can use it to slice and filter
+// and map and transform structured data with the same ease that sed, awk, grep
+// and friends let you play with text.
+//
+// See the [jq Tutorial](https://stedolan.github.io/jq/tutorial/) for more.
+//
+// This is a pure go implementation of jq. The type of v should be
+// []interface{} for an array or map[string]interface{} for a map.
+// Since the jq filter returns a iterator, it will be collected as a slice
+// even there is only one item.
+// It will insert the returned error message string as
+// the first and only item in the returned array.
+func jq(exp string, v any) []any {
```

Comment:
I guess you should change this function to return `([]any, error)`, see https://pkg.go.dev/text/template#FuncMap

Reply:
Returning `([]any, error)` is better. However I just saw the functions(like `fromYaml, toJson`) in `funcs.go` all inserting the error message into the normal returned value, besides the `TestFuncs()` use the error message to check the expectation. If we change the return values, how about introducing another test func?